metadata/src/main/java/org/apache/kafka/controller/errors/ControllerExceptions.java

```diff
@@ -38,4 +47,104 @@ public static boolean isTimeoutException(Throwable exception) {
         if (!(exception instanceof TimeoutException)) return false;
         return true;
     }
+
+    /**
+     * Check if an exception is a NotController exception.
+     *
+     * @param exception     The exception to check.
+     * @return              True if the exception is a NotController exception.
+     */
+    public static boolean isNotControllerException(Throwable exception) {
+        if (exception == null) return false;
+        if (exception instanceof ExecutionException) {
+            exception = exception.getCause();
+            if (exception == null) return false;
+        }
+        if (!(exception instanceof NotControllerException)) return false;
+        return true;
+    }
+
+    /**
+     * Create a new exception indicating that the controller is in pre-migration mode, so the
+     * operation cannot be completed.
+     *
+     * @param latestController  The current controller.
+     * @return                  The new NotControllerException.
+     */
+    public static NotControllerException newPreMigrationException(OptionalInt latestController) {
```

Comment:
I'm guessing the argument here is the controller ID? Can you rename it to reflect that? (same for other methods in this class)

Reply:
done